### PR TITLE
CB-10744 Fixing fragment link handling

### PR DESCRIPTION
**Pull request by dblotsky**
Adding support for name-based fragment anchors and anchors to IDs with special characters.

### CODE REVIEW

**Comment by dblotsky** [comment]
@riknoll @nikhilkh please review when you have a moment.

File: www/static/js/index.js [hunk 1]
```diff
@@ -145,17 +145,48 @@ $(document).ready(function () {
...
+        if (hash) {
 
             // prevent default anchor click behavior
             e.preventDefault();
 
-            // store hash
-            var hash = this.hash;
+            // get the fragment without the "#" symbol
+            var targetName = hash;
+            if (targetName[0] === "#") {
```
Comment (rakatyal) [inline www/static/js/index.js]: Do we need to return if this condition fails?

File: www/static/js/index.js [hunk 2]
```diff
@@ -145,17 +145,48 @@ $(document).ready(function () {
...
+                var targetName = targetName.slice(1);
```
Comment (nikhilkh) [inline www/static/js/index.js]: We should not re-define targetName with a `var`

File: www/static/js/index.js (hunk 1 above)
Comment (dblotsky) [inline www/static/js/index.js]: I guess the condition will never fail as long as browsers obey the standard. I'll remove the check.

**Comment by rakatyal** [comment]
LGTM.

File: www/static/js/index.js [hunk 3]
```diff
@@ -145,17 +145,46 @@ $(document).ready(function () {
...
+        if (hash) {
 
             // prevent default anchor click behavior
             e.preventDefault();
 
-            // store hash
-            var hash = this.hash;
+            // get the fragment without the "#" symbol because location.hash
+            // is returned with it
+            var targetName = targetName.slice(1);
```
Comment (riknoll) [inline www/static/js/index.js]: This line is a little confusing. Can you make the scoping of `targetName` clearer by renaming the variable?

Comment (dblotsky) [inline www/static/js/index.js]: Indeed. Turns out this line is also broken. Good catch! Fixed.

**Pull request merged**